### PR TITLE
Disable SMT for `ContractsAssets` and `ContractsState`

### PR DESCRIPTION
**Pull request by xgreenx**
Fixes https://github.com/FuelLabs/fuel-core/issues/2042

Added "smt" feature to the `fuel-core-storage` to enable old behavior. The SMT still is used in the tests and in benchmarks but disabled for the "production" mode.

## Checklist
- [x] Breaking changes are clearly marked as such in the PR description and changelog
- [x] New behavior is reflected in tests

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -139,3 +139,6 @@ itertools = { version = "0.12", default-features = false }
 insta = "1.8"
 tempfile = "3.4"
 tikv-jemallocator = "0.5"
+
+[patch.crates-io]
+fuel-vm = { path = "../fuel-vm/fuel-vm" }
```
Comment (Voxelot) [inline Cargo.toml:144]: ?

Comment (xgreenx) [reply Cargo.toml]: The PR requires https://github.com/FuelLabs/fuel-vm/pull/796 as well=)

File: Cargo.toml [hunk 2]
```diff
@@ -139,3 +139,6 @@ itertools = { version = "0.12", default-features = false }
 insta = "1.8"
 tempfile = "3.4"
 tikv-jemallocator = "0.5"
+
+[patch.crates-io]
+fuel-vm = { git = "https://github.com/FuelLabs/fuel-vm/", branch = "feature/implement-for-merkle-root-for-reference" }
```
Comment (MitchTurner) [inline Cargo.toml:144]: We still need to cut a new release of `fuel-vm` for this. It's merged though.

File: crates/services/executor/src/refs/contract.rs [hunk 3]
```diff
@@ -87,49 +88,101 @@ where
...
+    {
+        type InnerError = StorageError;
+    }
 }
 
 impl<'a, Database> ContractRef<&'a Database>
 where
-    Database: ContractStorageTrait,
-    anyhow::Error: From<Database::InnerError>,
+    &'a Database: ContractStorageTrait<InnerError = StorageError>,
```
[the remark this answers is not in the document]
Comment (MitchTurner) [reply crates/services/executor/src/refs/contract.rs:185]: I think this one is fine. In reality, it looks like we're just using the `ContractStorageTrait` as a sort of alias for all the other traits. I don't think we're using it anywhere else. But I'm not going to nitpick too much. I guess I'm just not a huge fan of the sort of inheritance-ness going on here.

**Review by MitchTurner — APPROVED** [review]

**Pull request merged**